react/features/deep-linking/functions.js

```diff
@@ -11,6 +11,11 @@ import {
 import { _shouldShowDeepLinkingDesktopPage }
     from './shouldShowDeepLinkingDesktopPage';
 
+/**
+ * Helper regex for replacing the URI protocol.
+ */
+const URI_PROTOCOL_REGEX = new RegExp(`^${URI_PROTOCOL_PATTERN}`);
```

Comment:
Why did you extract it here? RegExp usually has internal state. Not sure how string.replace works, can it cause issues?

Also it seems like this pattern is constantly prefixed with ^, so maybe it would be nice to just prefix the constant in its original place?

Reply:
I extracted it because I want to apply the same regex for matching in 2 places. It can't caue issues AFAIK.

> Also it seems like this pattern is constantly prefixed with ^, so maybe it would be nice to just prefix the constant in its original place?

Give that it's a **prefix** yeah, makes sense.